packages/website/ts/pages/governance/vote_form.tsx

```diff
@@ -0,0 +1,368 @@
+import styled from 'styled-components';
+
+// tslint:disable-next-line: no-duplicate-imports
+import { ChangeEvent, FormEvent } from 'react';
+
+import { colors } from 'ts/style/colors';
+
+import '@reach/dialog/styles.css';
+
+import { ContractWrappers } from '@0x/contract-wrappers';
+import { BigNumber, signTypedDataUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import { Button } from 'ts/components/button';
+import { Input } from 'ts/components/modals/input';
+import { Heading, Paragraph } from 'ts/components/text';
+import { PreferenceSelecter } from 'ts/pages/governance/preference_selecter';
+
+import { signatureUtils } from '0x.js';
+import * as ethUtil from 'ethereumjs-util';
+
+import { ECSignature, SignatureType } from '@0x/types';
+
+import { LedgerSubprovider } from '@0x/subproviders';
+import { Provider } from 'ethereum-types';
+import { InjectedProvider } from 'ts/types';
+import { configs } from 'ts/utils/configs';
+import { constants } from 'ts/utils/constants';
+import { utils } from 'ts/utils/utils';
+
+export enum VoteValue {
+    Yes = 'Yes',
+    No = 'No',
+}
+
+export interface VoteInfo {
+    userBalance: BigNumber;
+    voteValue: VoteValue;
+}
+
+interface Props {
+    onDismiss?: () => void;
+    onWalletConnected?: (providerName: string) => void;
+    onError?: (errorMessage: string) => void;
+    onVoted?: (userInfo: VoteInfo) => void;
+    web3Wrapper?: Web3Wrapper;
+    contractWrappers?: ContractWrappers;
+    currentBalance?: BigNumber;
+    selectedAddress: string;
+    isLedger: boolean;
+    injectedProvider?: InjectedProvider;
+    ledgerSubproviderIfExists?: LedgerSubprovider;
+    provider?: Provider;
+}
+
+interface State {
+    isWalletConnected: boolean;
+    isSubmitting: boolean;
+    isSuccessful: boolean;
+    isVoted: boolean;
+    selectedAddress?: string;
+    votePreference?: string;
+    zeip: number;
+    voteHash?: string;
+    signedVote?: SignedVote;
+    comment?: string;
+    errorMessage?: string;
+    errors: ErrorProps;
+}
+
+interface SignedVote {
+    signature: string;
+    from: string;
+    zeip: number;
+    preference: string;
+}
+
+interface FormProps {
+    isSuccessful?: boolean;
+    isSubmitting?: boolean;
+}
+
+interface ErrorProps {
+    [key: string]: string;
+}
+
+const defaultZeip = 23;
+
+export class VoteForm extends React.Component<Props> {
+    public static defaultProps = {
+        currentBalance: new BigNumber(0),
+        isWalletConnected: false,
+        isSubmitting: false,
+        isSuccessful: false,
+        isLedger: false,
+        isVoted: false,
+        zeip: defaultZeip,
+        errors: {},
+    };
+    public networkId: number;
+    public state: State = {
+        isWalletConnected: false,
+        isSubmitting: false,
+        isSuccessful: false,
+        isVoted: false,
+        votePreference: null,
+        zeip: VoteForm.defaultProps.zeip,
+        errors: {},
+    };
+    // shared fields
```

Comment:
Let's put a comment here about how the form is copied from our other form component (same as above)